Only in Package4.09_dist: Linux
diff -ru Package4.09_dist.orig/MCatNLO.Script Package4.09_dist/MCatNLO.Script
--- Package4.09_dist.orig/MCatNLO.Script	2012-09-04 12:23:51.000000000 +0200
+++ Package4.09_dist/MCatNLO.Script	2012-09-04 13:04:02.000000000 +0200
@@ -192,7 +192,9 @@
      else
        echo "no such option"; exit 1
      fi ;;
-MC) if [ $UsedPdfLib = "PDFLIB" ]
+MC)  echo "ERROR: MC step in CMSSW installation not supported."
+     exit
+     if [ $UsedPdfLib = "PDFLIB" ]
      then
        EXEC_NAMEF="MC_EXE_PDFLIB"
      elif [ $UsedPdfLib = "THISLIB" ]
@@ -762,6 +764,8 @@
 # wrapper for the functions which run and compile MC codes
 function runMC {
 compile MC
+echo "runMC not suported in this MC@NLO installation for CMSSW"
+exit
 whichherpdf $HERPDF
 if [ $pdftype = 1 ]
 then
@@ -1341,7 +1345,7 @@
   ExtraLibs="-l""$ExtraLibs"
 fi
 ##If using gfortran, reinstate the following
-##ExtraLibs=$ExtraLibs" -lstdc++"
+ExtraLibs=$ExtraLibs" -lstdc++ -lz"
 }
 
 # prepends -L to library paths
diff -ru Package4.09_dist.orig/MCatNLO.inputs Package4.09_dist/MCatNLO.inputs
--- Package4.09_dist.orig/MCatNLO.inputs	2012-09-04 12:23:51.000000000 +0200
+++ Package4.09_dist/MCatNLO.inputs	2012-09-04 13:27:11.000000000 +0200
@@ -246,7 +246,7 @@
 # prefix for the NLO and MC executables
 EXEPREFIX=ttb
 # number of events; set it to 0 to skip the event generation step
-NEVENTS=500000
+NEVENTS=100
 # set MCMODE=HW6 for HERWIG6 (Fortran) MCMODE=HWPP for HERWIG++
 MCMODE=HW6
 # 0 for weights=+1/-1, 1 for weights whose sum is the total rate
@@ -299,7 +299,8 @@
 # the local version of LHAPDF is installed. We assume that the library,
 # PDF sets, and configuration script are located in lib/, 
 # share/lhapdf/PDFsets/, and bin/ respectively
-LHALIBPATH="/home/frixione/Physics/lhapdf-5.8.5/lhapdf/"
+#LHALIBPATH="/afs/cern.ch/sw/lcg/external/MCGenerators/lhapdf/5.8.8/x86_64-slc5-gcc46-opt/"
+LHALIBPATH=`scramv1 tool info lhapdf | grep LHAPDF_BASE | sed 's/=/\n/g' | tail -1`
 # set LHAOFL=FREEZE to freeze PDFs from LHAPDF at the boundaries,
 # =EXTRAPOLATE otherwise. This variable is related to LHAPARM(18)
 LHAOFL=FREEZE
@@ -361,4 +362,4 @@
 # the following compiles MC code
 # compileMC
 
-runMCatNLO
+runNLO
diff -ru Package4.09_dist.orig/MCatNLO_pp.Script Package4.09_dist/MCatNLO_pp.Script
--- Package4.09_dist.orig/MCatNLO_pp.Script	2012-09-04 12:23:51.000000000 +0200
+++ Package4.09_dist/MCatNLO_pp.Script	2012-09-04 13:05:40.000000000 +0200
@@ -190,7 +190,9 @@
      else
        echo "no such option"; exit 1
      fi ;;
-MC) if [ $UsedPdfLib = "PDFLIB" ]
+MC)  echo "ERROR: MC step in CMSSW installation not supported."
+     exit
+     if [ $UsedPdfLib = "PDFLIB" ]
      then
        whichherpdf $HERPDF
        if [ $pdftype = 1 ]
@@ -282,6 +284,8 @@
    fi)
 # new compilation step for HW++ mode
 else
+  echo "runMC not suported in this MC@NLO installation for CMSSW"
+  exit
   # compile the analyzer & link it into the EXE directory
   # TODO: Stream via Inputs script
   FORFILEMAKE="FORFILE="$HWUTI
@@ -769,6 +773,8 @@
 # wrapper for the functions which run and compile MC codes
 # THIS IS NEW FOR HERWIG++ RUNNING
 function runMC {
+echo "runMC not suported in this MC@NLO installation for CMSSW"
+exit
 compile MC
 case $IPROC in
 -1706|-11706|1706|11706) runMCQQ ;;
@@ -1351,7 +1357,7 @@
   ExtraLibs="-l""$ExtraLibs"
 fi
 ##If using gfortran, reinstate the following
-##ExtraLibs=$ExtraLibs" -lstdc++"
+ExtraLibs=$ExtraLibs" -lstdc++ -lz"
 }
 
 # prepends -L to library paths
diff -ru Package4.09_dist.orig/Makefile Package4.09_dist/Makefile
--- Package4.09_dist.orig/Makefile	2012-09-04 12:23:51.000000000 +0200
+++ Package4.09_dist/Makefile	2012-09-04 14:14:56.000000000 +0200
@@ -1,60 +1,9 @@
-# Use this file as follows:
-# gmake -f Makefile EXTRAOBJ=<alpha,linux>.o VPATH=<vpath> <EXENAME>
-# See below for a list of name of executables. This is usually unnecessary,
-# the relevant operations being done by the scripts. If done manually, the
-# proper <vpath> must be entered at runtime
-
-ifeq ($(shell uname),AIX)
-F77=xlf -qextname -qflttrap=overflow:zerodivide:invalid:enable -O3 -qstrict \
-#       -qautodbl=dblpad
-SYSOBJ=
-AUTODBL=-qautodbl=dblpad
-endif
-ifeq ($(shell uname),SunOS)
-F77= f77 -fnonstd
-SYSOBJ=
-endif
-ifeq ($(shell uname),Linux)
-ifeq ($(COMPILERTYPE),xGFORTRAN)
-# Use with 
-#   F77= gfortran -w -O2 -fno-automatic 
-# has been tested successfully with IPROC=-16XX and exact mass depedence.
-F77= g77 -w -fno-automatic
-else
-F77= g77 -w -fno-automatic
-endif
+F77= gfortran -w -fno-automatic -fno-whole-file
 CPP= g++ $(INCLOPTION)
-SYSOBJ=trapfpe.o
-endif
-ifeq ($(shell uname),HP-UX)
-F77= g77 -w
-SYSOBJ=
-endif
-ifeq ($(shell uname),OSF1)
-F77= f77 
-CPP= g++ $(INCLOPTION)
-SYSOBJ=
-endif
-ifeq ($(shell uname),Darwin)
-ifeq ($(COMPILERTYPE),xGFORTRAN)
-F77= gfortran -w -O2 -fno-automatic
-else
-F77= g77 -w -fno-automatic
-endif
-CPP= g++ $(INCLOPTION)
-endif
 
 DEBUG=
 FF=$(F77) $(DEBUG)
 
-LIBS=`cernlib pdflib804 mathlib`
-#$LIBS gets replaced at compilation time by
-# /cern/pro/lib/libpdflib804.a /cern/pro/lib/libmathlib.a 
-# /cern/pro/lib/libpacklib.a -L/usr/local/lib -lshift -lnsl -lcrypt -ldl
-#on machines with a CERN installation. With a non-CERN installation,
-#replace the definition above with 
-#LIBS=/your/path/to/pdflib/libpdflib804.a .....
-
 %.o: $(SRCDIR)/%.f
 	$(F77) -I$(INCDIR) $(DEBUG) $(AUTODBL) -c $<
 %.o: $(COMSRC)/%.f
@@ -106,119 +55,40 @@
 HWFILES=mcatnlo_hwdriver.o mcatnlo_hwlhin.o \
         mcatnlo_str.o $(HWUTI)
 
-QQNLO_EXE_THISLIB : $(QQFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-QQNLO_EXE_PDFLIB : $(QQFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
 QQNLO_EXE_LHAPDF : $(QQFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-VVNLO_EXE_THISLIB : $(VBFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-VVNLO_EXE_PDFLIB : $(VBFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 VVNLO_EXE_LHAPDF : $(VBFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-HGNLO_EXE_THISLIB : $(HGFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-HGNLO_EXE_PDFLIB : $(HGFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 HGNLO_EXE_LHAPDF : $(HGFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-HGMNLO_EXE_THISLIB : $(HGMFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-HGMNLO_EXE_PDFLIB : $(HGMFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 HGMNLO_EXE_LHAPDF : $(HGMFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-SBNLO_EXE_THISLIB : $(SBFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-SBNLO_EXE_PDFLIB : $(SBFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 SBNLO_EXE_LHAPDF : $(SBFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-LLNLO_EXE_THISLIB : $(LLFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-LLNLO_EXE_PDFLIB : $(LLFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 LLNLO_EXE_LHAPDF : $(LLFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-VHNLO_EXE_THISLIB : $(VHFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-VHNLO_EXE_PDFLIB : $(VHFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 VHNLO_EXE_LHAPDF : $(VHFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-STNLO_EXE_THISLIB : $(STFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-STNLO_EXE_PDFLIB : $(STFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 STNLO_EXE_LHAPDF : $(STFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-WTDRNLO_EXE_THISLIB : $(WTDRFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-WTDRNLO_EXE_PDFLIB : $(WTDRFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 WTDRNLO_EXE_LHAPDF : $(WTDRFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-WTDSNLO_EXE_THISLIB : $(WTDSFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-WTDSNLO_EXE_PDFLIB : $(WTDSFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 WTDSNLO_EXE_LHAPDF : $(WTDSFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-HTDRNLO_EXE_THISLIB : $(HTDRFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-HTDRNLO_EXE_PDFLIB : $(HTDRFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 HTDRNLO_EXE_LHAPDF : $(HTDRFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
 HTDSNLO_EXE_THISLIB : $(HTDSFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
 	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-HTDSNLO_EXE_PDFLIB : $(HTDSFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-HTDSNLO_EXE_LHAPDF : $(HTDSFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-MC_EXE_THISLIB : $(HWFILES) $(HERWIGVER) $(PDFFILES)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-MC_EXE_PDFLIB : $(HWFILES) $(HERWIGVER) $(CPDFFILES) 
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-MC_EXE_LHAPDF : $(HWFILES) $(HERWIGVER) $(LPDFFILES)  
-	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
diff -ru Package4.09_dist.orig/Makefile_pp Package4.09_dist/Makefile_pp
--- Package4.09_dist.orig/Makefile_pp	2012-09-04 12:23:51.000000000 +0200
+++ Package4.09_dist/Makefile_pp	2012-09-04 12:29:00.000000000 +0200
@@ -1,60 +1,9 @@
-# Use this file as follows:
-# gmake -f Makefile_pp EXTRAOBJ=<alpha,linux>.o VPATH=<vpath> <EXENAME>
-# See below for a list of name of executables. This is usually unnecessary,
-# the relevant operations being done by the scripts. If done manually, the
-# proper <vpath> must be entered at runtime
-
-ifeq ($(shell uname),AIX)
-F77=xlf -qextname -qflttrap=overflow:zerodivide:invalid:enable -O3 -qstrict \
-#       -qautodbl=dblpad
-SYSOBJ=
-AUTODBL=-qautodbl=dblpad
-endif
-ifeq ($(shell uname),SunOS)
-F77= f77 -fnonstd
-SYSOBJ=
-endif
-ifeq ($(shell uname),Linux)
-ifeq ($(COMPILERTYPE),xGFORTRAN)
-# Use with 
-#   F77= gfortran -w -O2 -fno-automatic 
-# has been tested successfully with IPROC=-16XX and exact mass depedence.
-F77= g77 -w -fno-automatic
-else
-F77= g77 -w -fno-automatic
-endif
+F77= gfortran -w -fno-automatic
 CPP= g++ $(INCLOPTION)
-SYSOBJ=trapfpe.o
-endif
-ifeq ($(shell uname),HP-UX)
-F77= g77 -w
-SYSOBJ=
-endif
-ifeq ($(shell uname),OSF1)
-F77= f77 
-CPP= g++ $(INCLOPTION)
-SYSOBJ=
-endif
-ifeq ($(shell uname),Darwin)
-ifeq ($(COMPILERTYPE),xGFORTRAN)
-F77= gfortran -w -O2 -fno-automatic
-else
-F77= g77 -w -fno-automatic
-endif
-CPP= g++ $(INCLOPTION)
-endif
 
 DEBUG=
 FF=$(F77) $(DEBUG)
 
-LIBS=`cernlib pdflib804 mathlib`
-#$LIBS gets replaced at compilation time by
-# /cern/pro/lib/libpdflib804.a /cern/pro/lib/libmathlib.a 
-# /cern/pro/lib/libpacklib.a -L/usr/local/lib -lshift -lnsl -lcrypt -ldl
-#on machines with a CERN installation. With a non-CERN installation,
-#replace the definition above with 
-#LIBS=/your/path/to/pdflib/libpdflib804.a .....
-
 %.o: $(SRCDIR)/%.f
 	$(F77) -I$(INCDIR) $(DEBUG) $(AUTODBL) -c $<
 %.o: $(COMSRC)/%.f
@@ -96,101 +45,35 @@
 HWFILES=mcatnlo_hwdriver.o mcatnlo_hwlhin.o \
         mcatnlo_str.o $(HWUTI)
 
-QQNLO_EXE_THISLIB : $(QQFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-QQNLO_EXE_PDFLIB : $(QQFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 QQNLO_EXE_LHAPDF : $(QQFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-VVNLO_EXE_THISLIB : $(VBFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-VVNLO_EXE_PDFLIB : $(VBFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 VVNLO_EXE_LHAPDF : $(VBFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-HGNLO_EXE_THISLIB : $(HGFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-HGNLO_EXE_PDFLIB : $(HGFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 HGNLO_EXE_LHAPDF : $(HGFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-HGMNLO_EXE_THISLIB : $(HGMFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-HGMNLO_EXE_PDFLIB : $(HGMFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 HGMNLO_EXE_LHAPDF : $(HGMFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-LLNLO_EXE_THISLIB : $(LLFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-LLNLO_EXE_PDFLIB : $(LLFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 LLNLO_EXE_LHAPDF : $(LLFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-VHNLO_EXE_THISLIB : $(VHFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-VHNLO_EXE_PDFLIB : $(VHFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 VHNLO_EXE_LHAPDF : $(VHFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-STNLO_EXE_THISLIB : $(STFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-STNLO_EXE_PDFLIB : $(STFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 STNLO_EXE_LHAPDF : $(STFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-WTDRNLO_EXE_THISLIB : $(WTDRFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-WTDRNLO_EXE_PDFLIB : $(WTDRFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 WTDRNLO_EXE_LHAPDF : $(WTDRFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-WTDSNLO_EXE_THISLIB : $(WTDSFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-WTDSNLO_EXE_PDFLIB : $(WTDSFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 WTDSNLO_EXE_LHAPDF : $(WTDSFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-HTDRNLO_EXE_THISLIB : $(HTDRFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-HTDRNLO_EXE_PDFLIB : $(HTDRFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 HTDRNLO_EXE_LHAPDF : $(HTDRFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
 
-HTDSNLO_EXE_THISLIB : $(HTDSFILES) $(UTIFILES) $(PDFFILES) $(SYSOBJ)
-	$(FF) $^ $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
-HTDSNLO_EXE_PDFLIB : $(HTDSFILES) $(UTIFILES) $(CPDFFILES) $(SYSOBJ)
-	$(FF) $^ $(LIBS) $(EXTRAPATHS) $(EXTRALIBS) -o $@
-
 HTDSNLO_EXE_LHAPDF : $(HTDSFILES) $(LUTIFILES) $(LPDFFILES) $(SYSOBJ)
 	$(FF) $^ $(LHALIB) $(EXTRAPATHS) $(EXTRALIBS) -o $@
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_hgmain.f Package4.09_dist/srcHerwig6/mcatnlo_hgmain.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_hgmain.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_hgmain.f	2012-09-04 16:06:45.000000000 +0200
@@ -791,8 +791,8 @@
         call fk88strcat(fname,'_a',fnamea)
         call fk88strcat(fnamev,'_a.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         if(iwidth.eq.1)then
           call run_spring(sig5afw,fnamea,mx_of_evta,maxtrials,
      #                    nevts,ntrls,ndim,nwild,iseed)
@@ -821,8 +821,8 @@
         call fk88strcat(fname,'_b',fnameb)
         call fk88strcat(fnamev,'_b.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         if(iwidth.eq.1)then
           call run_spring(sig5bfw,fnameb,mx_of_evtb,maxtrials,
      #                    nevts,ntrls,ndim,nwild,iseed)
@@ -3231,6 +3231,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       write(iunit,902)ip1,ip2,ip3,ip4
       write(iunit,903)xevwgt,emsca
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_htmain_dr.f Package4.09_dist/srcHerwig6/mcatnlo_htmain_dr.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_htmain_dr.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_htmain_dr.f	2012-09-04 16:05:43.000000000 +0200
@@ -1212,8 +1212,8 @@
         call fk88strcat(fname,'_a',fnamea)
         call fk88strcat(fnamev,'_a.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+CC        call toend(22)
         call run_spring(sig5azw_ht,fnamea,mx_of_evta,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -1272,8 +1272,8 @@
         call fk88strcat(fname,'_b',fnameb)
         call fk88strcat(fnamev,'_b.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+CC        call toend(22)
         call run_spring(sig5bzw_ht,fnameb,mx_of_evtb,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -7223,6 +7223,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+CC      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6,ip7
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_htmain_ds.f Package4.09_dist/srcHerwig6/mcatnlo_htmain_ds.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_htmain_ds.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_htmain_ds.f	2012-09-04 16:07:07.000000000 +0200
@@ -1239,8 +1239,8 @@
         call fk88strcat(fname,'_a',fnamea)
         call fk88strcat(fnamev,'_a.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5azw_ht,fnamea,mx_of_evta,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -1299,8 +1299,8 @@
         call fk88strcat(fname,'_b',fnameb)
         call fk88strcat(fnamev,'_b.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5bzw_ht,fnameb,mx_of_evtb,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -7385,6 +7385,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6,ip7
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_llmain.f Package4.09_dist/srcHerwig6/mcatnlo_llmain.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_llmain.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_llmain.f	2012-09-04 16:07:22.000000000 +0200
@@ -755,8 +755,8 @@
         call fk88strcat(fname,'_a',fnamea)
         call fk88strcat(fnamev,'_a.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5afw,fnamea,mx_of_evta,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -779,8 +779,8 @@
         call fk88strcat(fname,'_b',fnameb)
         call fk88strcat(fnamev,'_b.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5bfw,fnameb,mx_of_evtb,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -4090,6 +4090,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       write(iunit,902)ip1,ip2,ip3,ip4,ip5
       write(iunit,903)xevwgt
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_qqmain.f Package4.09_dist/srcHerwig6/mcatnlo_qqmain.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_qqmain.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_qqmain.f	2012-09-04 16:07:38.000000000 +0200
@@ -1280,8 +1280,8 @@
            call fk88strcat(fname,'_a',fnamea)
            call fk88strcat(fnamev,'_a.events',fname1)
            open(unit=22,file=fname1,
-     #          form='formatted',status='old')
-           call toend(22)
+     #          form='formatted',access='append',status='old')
+C   C        call toend(22)
            call run_spring(sig5a,fnamea,mx_of_evta,maxtrials,
      #                     nevts,ntrls,ndim,nwild,iseed)
            close(22)
@@ -1340,8 +1340,8 @@
            call fk88strcat(fname,'_b',fnameb)
            call fk88strcat(fnamev,'_b.events',fname1)
            open(unit=22,file=fname1,
-     #          form='formatted',status='old')
-           call toend(22)
+     #          form='formatted',access='append',status='old')
+C   C        call toend(22)
            call run_spring(sig5b,fnameb,mx_of_evtb,maxtrials,
      #                     nevts,ntrls,ndim,nwild,iseed)
            close(22)
@@ -7100,6 +7100,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6,ip7,ip8,ip9
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_sbmain.f Package4.09_dist/srcHerwig6/mcatnlo_sbmain.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_sbmain.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_sbmain.f	2012-09-04 16:07:50.000000000 +0200
@@ -746,8 +746,8 @@
         call fk88strcat(fname,'_a',fnamea)
         call fk88strcat(fnamev,'_a.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         if(iwidth.eq.1)then
           call run_spring(sig5afw,fnamea,mx_of_evta,maxtrials,
      #                    nevts,ntrls,ndim,nwild,iseed)
@@ -775,8 +775,8 @@
         call fk88strcat(fname,'_b',fnameb)
         call fk88strcat(fnamev,'_b.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         if(iwidth.eq.1)then
           call run_spring(sig5bfw,fnameb,mx_of_evtb,maxtrials,
      #                    nevts,ntrls,ndim,nwild,iseed)
@@ -3670,6 +3670,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       write(iunit,902)ip1,ip2,ip3,ip4
       write(iunit,903)xevwgt
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_stmain.f Package4.09_dist/srcHerwig6/mcatnlo_stmain.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_stmain.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_stmain.f	2012-09-04 16:08:25.000000000 +0200
@@ -1151,8 +1151,8 @@
         call fk88strcat(fname,'_ain',fnameai)
         call fk88strcat(fnamev,'_ain.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5azw_in,fnameai,mx_of_evtai,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -1211,8 +1211,8 @@
         call fk88strcat(fname,'_aout',fnameao)
         call fk88strcat(fnamev,'_aout.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5azw_out,fnameao,mx_of_evtao,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -1271,8 +1271,8 @@
         call fk88strcat(fname,'_bin',fnamebi)
         call fk88strcat(fnamev,'_bin.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5bzw_in,fnamebi,mx_of_evtbi,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -1331,8 +1331,8 @@
         call fk88strcat(fname,'_bout',fnamebo)
         call fk88strcat(fnamev,'_bout.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5bzw_out,fnamebo,mx_of_evtbo,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -8313,6 +8313,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6,ip7
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_vbmain.f Package4.09_dist/srcHerwig6/mcatnlo_vbmain.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_vbmain.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_vbmain.f	2012-09-04 16:08:50.000000000 +0200
@@ -1099,8 +1099,8 @@
              call fk88strcat(fname,'_a',fnamea)
              call fk88strcat(fnamev,'_a.events',fname1)
              open(unit=22,file=fname1,
-     #            form='formatted',status='old')
-             call toend(22)
+     #            form='formatted',access='append',status='old')
+C   C        call toend(22)
              if(jproc.eq.3.and.prdct.ne.'ww')call swap(jproc)
              call run_spring(sig5a,fnamea,mx_of_evta(jproc),maxtrials,
      #                       nevts,ntrls,ndim,nwild,iseed)
@@ -1161,8 +1161,8 @@
              call fk88strcat(fname,'_b',fnameb)
              call fk88strcat(fnamev,'_b.events',fname1)
              open(unit=22,file=fname1,
-     #            form='formatted',status='old')
-             call toend(22)
+     #            form='formatted',access='append',status='old')
+C   C        call toend(22)
              if(jproc.eq.3.and.prdct.ne.'ww')call swap(jproc)
              call run_spring(sig5b,fnameb,mx_of_evtb(jproc),maxtrials,
      #                       nevts,ntrls,ndim,nwild,iseed)
@@ -8161,6 +8161,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6,ip7
@@ -8292,6 +8293,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6,ip7
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_vhmain.f Package4.09_dist/srcHerwig6/mcatnlo_vhmain.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_vhmain.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_vhmain.f	2012-09-04 16:05:44.000000000 +0200
@@ -871,7 +871,7 @@
         call fk88strcat(fnamev,'_a.events',fname1)
         open(unit=22,file=fname1,
      #       form='formatted',status='old')
-        call toend(22)
+C        call toend(22)
         if(iwidth.eq.1.and.idec.eq.0)then
           call run_spring(sig5afw_dec,fnamea,mx_of_evta,maxtrials,
      #                    nevts,ntrls,ndim,nwild,iseed)
@@ -914,7 +914,7 @@
         call fk88strcat(fnamev,'_b.events',fname1)
         open(unit=22,file=fname1,
      #       form='formatted',status='old')
-        call toend(22)
+C        call toend(22)
         if(iwidth.eq.1.and.idec.eq.0)then
           call run_spring(sig5bfw_dec,fnameb,mx_of_evtb,maxtrials,
      #                    nevts,ntrls,ndim,nwild,iseed)
@@ -5842,6 +5842,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_wtmain_dr.f Package4.09_dist/srcHerwig6/mcatnlo_wtmain_dr.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_wtmain_dr.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_wtmain_dr.f	2012-09-04 16:09:07.000000000 +0200
@@ -1203,8 +1203,8 @@
         call fk88strcat(fname,'_a',fnamea)
         call fk88strcat(fnamev,'_a.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5azw_wt,fnamea,mx_of_evta,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -1263,8 +1263,8 @@
         call fk88strcat(fname,'_b',fnameb)
         call fk88strcat(fnamev,'_b.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5bzw_wt,fnameb,mx_of_evtb,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -7122,6 +7122,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6,ip7,ip8
diff -ru Package4.09_dist.orig/srcHerwig6/mcatnlo_wtmain_ds.f Package4.09_dist/srcHerwig6/mcatnlo_wtmain_ds.f
--- Package4.09_dist.orig/srcHerwig6/mcatnlo_wtmain_ds.f	2012-09-04 12:23:52.000000000 +0200
+++ Package4.09_dist/srcHerwig6/mcatnlo_wtmain_ds.f	2012-09-04 16:09:22.000000000 +0200
@@ -1225,8 +1225,8 @@
         call fk88strcat(fname,'_a',fnamea)
         call fk88strcat(fnamev,'_a.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5azw_wt,fnamea,mx_of_evta,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -1285,8 +1285,8 @@
         call fk88strcat(fname,'_b',fnameb)
         call fk88strcat(fnamev,'_b.events',fname1)
         open(unit=22,file=fname1,
-     #       form='formatted',status='old')
-        call toend(22)
+     #       form='formatted',access='append',status='old')
+C        call toend(22)
         call run_spring(sig5bzw_wt,fnameb,mx_of_evtb,maxtrials,
      #                  nevts,ntrls,ndim,nwild,iseed)
         close(22)
@@ -7278,6 +7278,7 @@
         write(*,*)'Fatal error in store_events: xpmone=',xpmone
         stop
       endif
+C      backspace(iunit)
       write(iunit,901)i1hpro,iccode,np
       if(idec.eq.0)then
         write(iunit,902)ip1,ip2,ip3,ip4,ip5,ip6,ip7,ip8
